### PR TITLE
Move ideal_storage_size calculation before it is used

### PR DESCRIPTION
**Pull request by dmakarov**
#### Problem

The `ideal_storage_size` tuning parameter is incorrectly calculated after it needs to be used.

#### Summary of Changes

Move the update of the `ideal_storage_size` tuning parameter to an earlier point in the control flow before it is used.

Adjust tests.

### CODE REVIEW

File: accounts-db/src/ancient_append_vecs.rs [hunk 1]
```diff
@@ -522,9 +516,16 @@ impl AccountsDb {
     fn collect_sort_filter_ancient_slots(
         &self,
         slots: Vec<Slot>,
-        tuning: &PackedAncientStorageTuning,
+        tuning: &mut PackedAncientStorageTuning,
     ) -> AncientSlotInfos {
         let mut ancient_slot_infos = self.calc_ancient_slot_info(slots, tuning);
+        // ideal storage size is total alive bytes of ancient storages
```
Comment (HaoranYi) [inline accounts-db/src/ancient_append_vecs.rs:528]: Can we add a test for this, i.e. after we call collect_sort_filter_ancinet_slots() make sure that ideal_storage_size is updated?

**Review by brooksprumo — COMMENTED** [review]
overall looks good to me. Please re-request a review when PR is ready for review again/has the test. Thanks!

**Comment by jeffwashington** [comment]
definitely looks like it is working:
image

it used to write 130GB each time, now it writes 57MB (5.7MB * 10).
We may still want to adjust the 10 constant. But, at least we are respecting the limit correctly now.

**Comment by dmakarov** [comment]
> definitely looks like it is working: image
> 
> it used to write 130GB each time, now it writes 57MB (5.7MB * 10). We may still want to adjust the 10 constant. But, at least we are respecting the limit correctly now.

that's good. the constant is for another PR.

**Review by brooksprumo — APPROVED** [review]
:shipit:

**Review by jeffwashington — APPROVED** [review]
lgtm

**Review by HaoranYi — APPROVED** [review]
lgtm.

Thanks for adding the test.

**Pull request merged**